dbos/_dbos.py

```diff
@@ -334,6 +339,36 @@ def __init__(
         self.conductor_key: Optional[str] = conductor_key
         self.conductor_websocket: Optional[ConductorWebsocket] = None
 
+        init_logger()
+
+        unvalidated_config: Optional[ConfigFile] = None
+
+        if config is None:
+            # If no config is provided, load it from dbos-config.yaml
+            unvalidated_config = load_config(run_process_config=False)
+        elif is_dbos_configfile(config):
+            unvalidated_config = cast(ConfigFile, config)
+            if os.environ.get("DBOS__CLOUD") == "true":
+                unvalidated_config = overwrite_config(unvalidated_config)
```

Comment:
If the `config` value was a `ConfigFile` in the first place, why does it need to be overwritten? Aren't we loading twice from the same file in this case?

Reply:
On this execution path, the user provides a `ConfigFile` struct to the `DBOS` constructor (the values are not taken from the configuration file)

However, in DBOS Cloud, we force some configuration options and set them in the config file. So we need a way to overwrite user-provided input for config file values at runtime. This is what this is doing.